### DISCUSSION TITLE
findMany returns all records when condition leaks as undefined

### DISCUSSION THREAD

**Discussion by SharadKumar** [prisma/prisma#3386] [Q&A] [answered]
if `accountId` is undefined (lets say it is coming from querystring)

```ts
const products = await prisma.product.findMany({
            where: {
              accountId: {
                equals: accountId as string,
              },
            },
          })
```

Is it normal or expected behaviour, I wonder?!!
Of course I can handle it my side, but then I have to handle it in all usages. I am wondering if this should be handled at the prisma level? Whats the best practice?

**Comment by ryands17** [comment, marked as answer]
Hey @SharadKumar 👋 
This is expected behaviour. Whenever you write the query like this:

```ts
const products = await prisma.product.findMany({
  where: {},
})
```

The `accountId` is entirely optional and implicitly undefined. So the above will be equivalent to this when `accountId` is `undefined`:

```ts
const products = await prisma.product.findMany({
  where: {
    accountId: { equals: undefined }, 
  },
})
```

This is just the underlying behaviour of JavaScript and so you would need to handle this in your logic itself.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
